### PR TITLE
packageId -> updateId in ChunksReceived

### PR DESCRIPTION
**Pull request by rdanitz**
query update related entities like package when needed

### CODE REVIEW

**Comment by rdanitz** [comment]
@jerrytrieu this still looks way more complicated than it should be, especially when disregarding dependencies. maybe some pruning?

File: core/src/main/scala/org/genivi/sota/core/rvi/TransferController.scala [hunk 1]
```diff
@@ -127,42 +126,63 @@ class TransferProtocolActor(db: Database, rviClient: RviClient,
...
+  case class PackageNotFound(packageId: Package.Id)
+    extends Throwable(s"Package $packageId not found!") with NoStackTrace
+
+  case class UpdateNotFound(updateId: UUID)
+      extends Throwable(s"Update $updateId not found!") with NoStackTrace
+
+  def startPackageUpload(services: ClientServices)
+                        (updateId: UUID): Future[(ActorRef, UUID)] = {
+
+    db.run(UpdateRequests.byId(updateId)).flatMap(_ match {
```
Comment (jerrytrieu) [inline core/src/main/scala/org/genivi/sota/core/rvi/TransferController.scala]: Is this extra db query?

File: core/src/main/scala/org/genivi/sota/core/rvi/TransferController.scala [hunk 2]
```diff
@@ -127,42 +126,63 @@ class TransferProtocolActor(db: Database, rviClient: RviClient,
...
-        val finishedPackageTransfers = done + p
-        val (finishedUpdates, unfinishedUpdates) = updates.span(_.dependencies.diff(finishedPackageTransfers).isEmpty)
-        context.become(
-          running( services, unfinishedUpdates, if(pending.isEmpty) pending else pending.tail,
-                   newInProgress, finishedPackageTransfers) )
+      val oldId = inProgress(ref)
+      log.debug(s"Package for update $oldId uploaded.")
+
+      pending.headOption match {
+        case Some(currentId) =>
```
Comment (jerrytrieu) [inline core/src/main/scala/org/genivi/sota/core/rvi/TransferController.scala]: Add comment that not current code doesn't expect non-empty 'pending'?

**Comment by jerrytrieu** [comment]
@rdanitz cool, works now

**Comment by rdanitz** [comment]
superseded by #303

**Pull request closed without merging**